### ISSUE TITLE
query via Lucene returns incorrect results

### ISSUE THREAD

**Issue by rdelangh** [orientechnologies/orientdb#6848]
## OrientDB Version, operating system, or hardware.
- v2.2.12 SNAPSHOT[x] - 20161019

## Operating System
- [x] Linux

Records (documents) get INSERTed into a class; on a property FILENAME there is a FULLTEXT LUCENE index created.

When I search in this class for records that have a certain FILENAME, via the LUCENE operator (forcing it to use the Lucene index), the result is incorrect compared to the use of operator "=":
```
SELECT count(*) FROM cdr_icx_20161023 WHERE ENV='uat' AND FILENAME = 'ICX001ICX20161025100000.109444'
--> returns 8
SELECT count(*) FROM cdr_icx_20161023 WHERE ENV='uat' AND FILENAME LUCENE 'ICX001ICX20161025100000.109444'
--> return 1
```

=> Lucene index is incorrect.

**Comment by rdelangh** [comment]
Another example from another class, which also has a FULLTEXT LUCENE index on its property FILENAME:
```
SELECT count(*) FROM cdr_af_20161023 WHERE ENV='uat' AND FILENAME = 'CTS020IMS20161025131804.004893.raw.C22N251314000012.001.00025518'
--> returns 3
SELECT count(*) FROM cdr_af_20161023 WHERE ENV='uat' AND FILENAME LUCENE 'CTS020IMS20161025131804.004893.raw.C22N251314000012.001.00025518'
--> returns 1
```

**Comment by maggiolo00** [comment]
@rdelangh 

do you have an index defined on EVN property?

**Comment by maggiolo00** [comment]
@rdelangh 
If you rebuild the index on cdr_af_20161023.FILENAME ?

**Comment by rdelangh** [comment]
@maggiolo00 
yes, there is a second index which includes ENV.

This query returns incorrect result 1 instead of 10:
```
SELECT count(*) FROM cdr_icx_20161023 WHERE ENV='uat' AND FILENAME LUCENE 'ICX003ICX20161025100000.109450'
```

However this query returns correct result 10:
```
SELECT count(*) FROM cdr_icx_20161023 WHERE FILENAME LUCENE 'ICX003ICX20161025100000.109450'
```

-> seems buggy behaviour ...

**Comment by maggiolo00** [comment]
@rdelangh 

can you post the explain of 

```
SELECT count(*) FROM cdr_icx_20161023 WHERE ENV='uat' AND FILENAME LUCENE 'ICX003ICX20161025100000.109450'
```

Thanks

**Comment by rdelangh** [comment]
@maggiolo00 

Here you go:
```
{
    "result": [
        {
            "@type": "d",
            "@version": 0,
            "documentReads": 1,
            "fullySortedByIndex": false,
            "documentAnalyzedCompatibleClass": 1,
            "recordReads": 1,
            "fetchingFromTargetElapsed": 1,
            "indexIsUsedInOrderBy": false,
            "current": "#852:397361",
            "_memoryIndex": "FILENAME:\n\t'[69 63 78 30 30 33 69 63 78 32 30 31 36 31 30 32 35 31 30 30 30 30 30 2e 31 30 39 34 35 30]':1: [(0)]\n\tterms=1, positions=1\ncdr_icx_20161023.FILENAME:\n\t'[69 63 78 30 30 33 69 63 78 32 30 31 36 31 30 32 35 31 30 30 30 30 30 2e 31 30 39 34 35 30]':1: [(0)]\n\tterms=1, positions=1\n\nfields=2, terms=2, positions=2",
            "involvedIndexes": [
                "idx_cdr_icx_20161023_0"
            ],
            "projectionElapsed": 0,
            "limit": -1,
            "evaluated": 1,
            "groupByElapsed": 0,
            "user": "#5:0",
            "elapsed": 19.436144,
            "resultType": "collection",
            "resultSize": 1,
            "@fieldTypes": "documentReads=l,documentAnalyzedCompatibleClass=l,recordReads=l,fetchingFromTargetElapsed=l,current=x,involvedIndexes=e,projectionElapsed=l,evaluated=l,groupByElapsed=l,user=x,elapsed=f"
        }
    ],
    "notification": "Query executed in 0.052 sec. Returned 1 record(s)"
}
```

**Comment by rdelangh** [comment]
The index "idx_cdr_icx_20161023_0" is a UNIQUE (not Lucene) index made of ENV, PRT_DATE (datetime), FILENAME and RECORD_NUMBER

Another index "idx_cdr_icx_20161023_1" is the LUCENE on FILENAME

**Comment by maggiolo00** [comment]
@rdelangh 

so in this query the idx_cdr_icx_20161023_0 is used then the lucene operator is use for in memory evaluation.

can you post also the explain of 

```
SELECT count(*) FROM cdr_icx_20161023 WHERE ENV='uat' AND FILENAME = 'ICX003ICX20161025100000.109450'
```

**Comment by rdelangh** [comment]
@maggiolo00 

Here you go : 
```
{
    "result": [
        {
            "@type": "d",
            "@version": 0,
            "documentReads": 10,
            "fullySortedByIndex": false,
            "documentAnalyzedCompatibleClass": 10,
            "recordReads": 10,
            "fetchingFromTargetElapsed": 1,
            "indexIsUsedInOrderBy": false,
            "compositeIndexUsed": 1,
            "current": "#851:397360",
            "involvedIndexes": [
                "idx_cdr_icx_20161023_0"
            ],
            "projectionElapsed": 0,
            "limit": -1,
            "evaluated": 10,
            "groupByElapsed": 0,
            "user": "#5:0",
            "elapsed": 1.405957,
            "resultType": "collection",
            "resultSize": 1,
            "@fieldTypes": "documentReads=l,documentAnalyzedCompatibleClass=l,recordReads=l,fetchingFromTargetElapsed=l,compositeIndexUsed=l,current=x,involvedIndexes=e,projectionElapsed=l,evaluated=l,groupByElapsed=l,user=x,elapsed=f"
        }
    ],
    "notification": "Query executed in 0.056 sec. Returned 1 record(s)"
}
```

**Comment by maggiolo00** [comment]
@rdelangh 

can you reproduce this on a small dataset with a script?

**Comment by maggiolo00** [comment]
@rdelangh 

how did you create the lucene index?

**Comment by maggiolo00** [comment]
can you also post

the explain of 

```
explain SELECT FROM cdr_icx_20161023 WHERE ENV='uat' AND FILENAME LUCENE 'ICX003ICX20161025100000.109450'
```

```
explain SELECT FROM cdr_icx_20161023 WHERE ENV='uat' AND FILENAME LUCENE 'ICX003ICX20161025100000.109450'
```

adn 

```
explain SELECT FROM cdr_icx_20161023 WHERE ENV='uat' AND FILENAME = 'ICX003ICX20161025100000.109450'
```

**Comment by maggiolo00** [comment]
@rdelangh 

i think i was able to reproduce that. I'm working on a fix.

Thanks

**Comment by rdelangh** [comment]
@maggiolo00 
ok, good news !

**Comment by maggiolo00** [comment]
@rdelangh  fixed. 
It will be available on 2.2.13 GA or  you can try the snapshot

Thanks
